--
-- PostgreSQL database dump
--

-- Dumped from database version 17.5
-- Dumped by pg_dump version 17.2

SET statement_timeout = 0;
SET lock_timeout = 0;
SET idle_in_transaction_session_timeout = 0;
SET transaction_timeout = 0;
SET client_encoding = 'UTF8';
SET standard_conforming_strings = on;
SELECT pg_catalog.set_config('search_path', '', false);
SET check_function_bodies = false;
SET xmloption = content;
SET client_min_messages = warning;
SET row_security = off;

--
-- Data for Name: t_scripts; Type: TABLE DATA; Schema: sw; Owner: d3l243
--

COPY sw.t_scripts (script_id, script, description, enabled, results_tag, contents, parameters, fields, backfill_to_dms, pipeline_job_enabled, pipeline_mac_job_enabled) FROM stdin;
0	(none)	"Standard" processing for DMS tool	N	\N	\N	\N	\N	0	0	0
1	Sequest	This is the standard script for the "Sequest" DMS tool (modified to have conditional data extraction dependency).  Includes MSGF and IDPicker. Retired in 2021.	N	Seq	<JobScript Name="Sequest"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="Sequest"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="IDPicker"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
2	ICR2LS	Deisotope data using ICR-2LS.  Retired in 2021.	N	ICR	<JobScript Name="ICR2LS"><Step Number="1" Tool="ICR2LS" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>			0	0	0
4	TIC_ICR	Create TICs using ICR-2LS.  Retired after MASIC was developed.	N	TIC	<JobScript Name="TIC_ICR"><Step Number="1" Tool="Bogus" /></JobScript>			0	0	0
5	TIC_LCQ	"Standard" processing for DMS tool	N	TIC	\N	\N	\N	0	0	0
6	QTOFSequest	"Standard" processing for DMS tool	N	\N	\N	\N	\N	0	0	0
7	QTOFPek	"Standard" processing for DMS tool	N	\N	\N	\N	\N	0	0	0
8	DeNovoID	"Standard" processing for DMS tool	N	\N	\N	\N	\N	0	0	0
9	AgilentSequest	"Standard" processing for DMS tool	N	Seq	\N	\N	\N	0	0	0
10	MLynxPek	"Standard" processing for DMS tool	N	ICR	\N	\N	\N	0	0	0
11	AgilentTOFPek	"Standard" processing for DMS tool	N	ICR	\N	\N	\N	0	0	0
12	LTQ_FTPek	"Standard" processing for DMS tool	N	ICR	<JobScript Name="LTQ_FTPek"><Step Number="1" Tool="LTQ_FTPek" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>			0	0	0
13	MASIC_Finnigan	This is the standard script for the "MASIC_Finnigan" DMS tool	Y	SIC	<JobScript Name="MASIC_Finnigan"><Step Number="1" Tool="MASIC_Finnigan" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>	\N	\N	0	0	0
14	MASIC_Agilent	This is the standard script for the "MASIC_Agilent" DMS tool	N	SIC	<JobScript Name="MASIC_Agilent"><Step Number="1" Tool="MASIC_Agilent" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>			0	0	0
15	XTandem	This is the standard script for the "XTandem" DMS tool	Y	XTM	<JobScript Name="XTandem"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="XTandem"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="IDPicker"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
16	Decon2LS	This is the standard script for the "Decon2LS" DMS tool.	N	DLS	<JobScript Name="Decon2LS"><Step Number="1" Tool="Decon2LS" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>	\N	\N	0	0	0
17	TIC_D2L	This is the standard script for the "TIC_D2L" DMS tool	N	TDL	<JobScript Name="TIC_D2L"><Step Number="1" Tool="TIC_D2L" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>			0	0	0
18	Decon2LS_Agilent	This is the standard script for the "Decon2LS_Agilent" DMS tool	N	DLS	<JobScript Name="Decon2LS_Agilent"><Step Number="1" Tool="Decon2LS_Agilent" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>	\N	\N	0	0	0
19	TIC_D2L_Agilent	This is the standard script for the "TIC_D2L_Agilent" DMS tool	N	TDL	<JobScript Name="TIC_D2L_Agilent"><Step Number="1" Tool="TIC_D2L_Agilent" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>			0	0	0
20	Inspect	This is the parallelized script for the "Inspect" DMS tool; retired in 2021	N	INS	<JobScript Name="Inspect"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="DTA_Split"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="Inspect" Special="Clone"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="InspectResultsAssembly"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="DataExtractor"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="MSGF"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>			0	0	0
21	XTandemSplice	This is a splice script to add XTandem steps onto standard Sequest script job	N	XTS	<JobScript Name="XTandemSplice"><Step Number="10" Tool="XTandem"><Depends_On Step_Number="2" /></Step><Step Number="11" Tool="DataExtractor"><Depends_On Step_Number="10" /></Step><Step Number="12" Tool="Results_Transfer"><Depends_On Step_Number="11" /></Step></JobScript>			0	0	0
22	MSXML_Gen	Uses ReadW or MSConvert to create mzXML or mzML files.	Y	XML	<JobScript Name="MSXML_Gen"><Step Number="1" Tool="MSXML_Gen" Special="Job_Results" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>	\N	\N	0	0	0
23	DTA_Gen	Uses Extract_MSn or DeconMSn to create a concatenated DTA (_Dta.txt) file	Y	DTA	<JobScript Name="DTA_Gen"><Step Number="1" Tool="DTA_Gen" Special="Job_Results" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>	\N	\N	0	0	0
24	MSClusterDAT_Gen	This script generates .dat files for use by Ari Frank's MSCluster tool	N	MSCD	<JobScript Name="MSClusterDAT_Gen"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSClusterDTAtoDAT"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="2" Enable_Only="1" /></Step><Step Number="4" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>	\N	\N	0	0	0
25	DTA_Import	Imports externally-supplied DTA folders into a dataset for later analysis	N	DTA	<JobScript Name="DTA_Import"><Step Number="1" Tool="DTA_Import" Special="Job_Results" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>	\N	\N	0	0	0
26	XTandem_DTA_Import	This is the standard script for the "XTandem" DMS tool, except that it can do DTA import also	N	XTM	<JobScript Name="XTandem"><Step Number="1" Tool="DTA_Gen" /><Step Number="7" Tool="DTA_Import"><Depends_On Step_Number="1" /></Step><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /><Depends_On Step_Number="7" Test="No_Parameters" /></Step><Step Number="3" Tool="XTandem"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="7" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step></JobScript>	\N	\N	0	0	0
27	Sequest_UseExistingExternalDTA	This script runs the "Sequest" DMS tool, but assumes that an External DTA folder already exist in the Dataset folder; intended for use with FinniganDefSettings_ExternalDTA.xml or a similar settings file.  The settings file should have an entry like: item key="ExternalDTAFolderName" value="DTA_Manual_01". Retired in 2021.	N	Seq	<JobScript Name="Sequest"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="Sequest"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
28	XTandem_UseExistingExternalDTA	This script runs the "X!Tandem" DMS tool, but assumes that an External DTA folder already exist in the Dataset folder; intended for use with FinniganDefSettings_ExternalDTA.xml or a similar settings file.  The settings file should have an entry like: item key="ExternalDTAFolderName" value="DTA_Manual_01". Retired in 2021.	N	XTM	<JobScript Name="XTandem"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="XTandem"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>			0	0	0
29	Decon2LS_V2	This script runs the DeconToolsAutoProcessor, which supports IMS data and RAPID.  Following Decon2LS, the LCMSFeatureFinder processes the Decon2LS results.  The Settings file must contain an LCMSFeatureFinder section (with LCMSFeatureFinderIniFile defining the Feature Finder .Ini file to use)	Y	DLS	<JobScript Name="Decon2LS_V2"><Step Number="1" Tool="Decon2LS_V2" /><Step Number="2" Tool="LCMSFeatureFinder"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>	\N	\N	0	0	0
30	OMSSA	This is the standard script for the "OMSSA" DMS tool	N	OMS	<JobScript Name="OMSSA"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="OMSSA"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="4" Enable_Only="1" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="4" /></Step></JobScript>	\N	\N	0	0	0
31	LCMSFeatureFinder_UseExistingResults	This script runs the LCMSFeatureFinder against existing Decon2LS results.  The Settings file must contain an LCMSFeatureFinder section (with LCMSFeatureFinderIniFile defining the Feature Finder .Ini file to use)	Y	DLS	<JobScript Name="Decon2LS"><Step Number="3" Tool="LCMSFeatureFinder"><Depends_On Step_Number="1" /></Step><Step Number="4" Tool="Results_Transfer"><Depends_On Step_Number="3" /></Step></JobScript>	\N	\N	0	0	0
32	Sequest_DTARefinery	This script uses DTARefinery to refine the DTAs before running "Sequest" (conditional data extraction dependency).  Includes MSGF and IDPicker. Retired in 2021.	N	Seq	<JobScript Name="Sequest"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="DTA_Refinery"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="Sequest"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="IDPicker"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="10" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>	\N	\N	0	0	0
33	XTandem_HPC	This script runs "XTandem" on the Chinook high-performance computer; retired in 2021	N	XTM	<JobScript Name="XTandem"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="DTA_Split"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="XTandem_HPC"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>			0	0	0
34	XTandem_DTARefinery	This script uses DTARefinery to refine the DTAs before running "XTandem" (conditional data extraction dependency); retired in 2021	N	XTM	<JobScript Name="XTandem"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="DTA_Refinery"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="XTandem"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="IDPicker"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="10" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>			0	0	0
35	Inspect_UseExistingExternalDTA	This script runs "Inspect", but assumes that an External DTA folder already exists in the Dataset folder; intended for use with Inspect_ExternalDTA_x1.xml or a similar settings file.  The settings file should have an entry like: item key="ExternalDTAFolderName" value="DTA_Manual_01"; retired in 2021	N	INS	<JobScript Name="Inspect"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="DTA_Split"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="Inspect" Special="Clone"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="InspectResultsAssembly"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="DataExtractor"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="MSGF"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>			0	0	0
36	PRIDE_mzXML	Aggregates mzXML results from multiple mzXML jobs and saves result in data package	N	PZX	<JobScript Name="PRIDE_mzXML"><Step Number="1" Tool="mzXML_Aggregator" /></JobScript>	<Param Section="JobParameters" Name="DatasetName" Value="Aggregation" /><Param Section="JobParameters" Name="PRIDEMzXMLInputFile" Value="" /><Param Section="JobParameters" Name="TargetJobFileList" Value="sequest:_syn.txt:copy,msxml_gen:.mzxml:nocopy" /><Param Section="JobParameters" Name="transferFolderPath" Value="" />	\N	1	0	0
37	Phospho_FDR_Aggregator	Run AScore on a series of MSGF+ results defined in a data package	Y	PFA	<JobScript Name="Phospho_FDR_Aggregator"><Step Number="1" Tool="Phospho_FDR_Aggregator" /></JobScript>	<Param Section="JobParameters" Name="AScoreCIDParamFile" Value="AScore_CID_0.5Da_ETD_0.5Da_HCD_0.05Da.xml" Reqd="No" /><Param Section="JobParameters" Name="AScoreETDParamFile" Value="xxx_Undefined.xml" Reqd="No" /><Param Section="JobParameters" Name="AScoreHCDParamFile" Value="xxx_Undefined.xml" Reqd="No" /><Param Section="JobParameters" Name="DatasetName" Value="Aggregation" Reqd="Yes" /><Param Section="JobParameters" Name="TargetJobFileList" Value="MSGFPlus:_msgfplus_syn.txt,MSGFPlus:_msgfplus_syn_ModSummary.txt,MSGFPlus:_dta.zip" Reqd="Yes" /><Param Section="JobParameters" Name="transferFolderPath" Value="" Reqd="No" />	\N	1	1	0
38	Sequest_Extend_MSGF	This script should be used to process existing Sequest results to process them with MSGF. Retired in 2021.	N	Seq	<JobScript Name="Sequest_AddMSGFToExisting"><Step Number="7" Tool="MSGF"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>	\N	\N	0	0	0
39	Sequest_No_MSGF	This is the standard script for the "Sequest" DMS tool.  This applies to jobs that ran sequest but did not use MSGF. Retired in 2021.	N	Seq	<JobScript Name="Sequest"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="Sequest"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="4" Enable_Only="1" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="4" /></Step></JobScript>	\N	\N	0	0	0
40	LCMSFeature_Finder	This script runs the LCMSFeatureFinder to process the Decon2LS results.  This is a case where a job targets another job.  Use the text SourceJob:0000 in the job's Special_Processing parameters to define the parent DeconTools job.  Alternatively, use a job match spec, e.g.: SourceJob:Auto{Tool = "Decon2LS_V2" AND Settings_File = "Decon2LS_FF_IMS_0.15Fit_0.3IScore_20ppm_500Int_DaCorrect_2011-01-24.xml" AND Param_File = "IMS_UIMF_PeakBR4_PeptideBR4_SN3_SumScans4_SumFrames3_noFit_Thrash_2009-11-02.xml"}	Y	LFF	<JobScript Name="LCMSFeature_Finder"><Step Number="1" Tool="LCMSFeatureFinder" Special="ExtractSourceJobFromComment" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>			0	0	0
41	XTandem_No_MSGF	This is the standard script for the "XTandem" DMS tool. This applies to jobs that ran X!Tandem but did not use MSGF	Y	XTM	<JobScript Name="XTandem"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="XTandem"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="4" Enable_Only="1" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="4" /></Step></JobScript>	\N	\N	0	0	0
42	XTandem_Extend_MSGF	This script should be used to process existing X!Tandem results to process them with MSGF	Y	XTM	<JobScript Name="XTandem_AddMSGFToExisting"><Step Number="7" Tool="MSGF"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>	\N	\N	0	0	0
43	Inspect_Single	This is the non-parallelized script for the "Inspect" DMS tool; retired in 2021	N	INS	<JobScript Name="Inspect"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="Inspect"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="InspectResultsAssembly"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>			0	0	0
44	Sequest_DTARefinery_No_MSGF	This script uses DTARefinery to refine the DTAs before running "Sequest" (conditional data extraction dependency); this was used prior to MSGF. Retired in 2021.	N	Seq	<JobScript Name="Sequest"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="DTA_Refinery"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="Sequest"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
45	Sequest_DTARefinery_Extend_MSGF	This script should be used to process existing Sequest DTARefinery results to process them with MSGF. Retired in 2021.	N	Seq	<JobScript Name="Sequest_DTARefinery_AddMSGFToExisting"><Step Number="9" Tool="MSGF"><Depends_On Step_Number="8" /></Step><Step Number="10" Tool="Results_Transfer"><Depends_On Step_Number="9" /></Step></JobScript>	\N	\N	0	0	0
46	MSXML_Bruker	Uses the CompassXport tool to creates mzXML or mzML files from Bruker .D folders	Y	XML	<JobScript Name="MSXML_Bruker"><Step Number="1" Tool="MSXML_Bruker" Special="Job_Results" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>	\N	\N	0	0	0
47	MultiAlign	This script runs MultiAlign on a single job against a given AMT tag database.  Specify the AMT tag database to use using the Special_Processing parameters, e.g. AMTDB:MT_Shewanella_ProdTest_Formic_P460.  Use the text SourceJob:0000 in the job's Special_Processing parameters to define the parent DeconTools job.  Alternatively, use a job match spec, e.g.: SourceJob:Auto{Tool = "Decon2LS_V2" ORDER BY Job Desc}.	N	MAL	<JobScript Name="MultiAlign"><Step Number="1" Tool="MultiAlign" Special="ExtractSourceJobFromComment" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>			0	0	0
48	MultiAlign_Aggregator	This script runs MultiAlign on a set of jobs (optionally align to an AMT tag database).  Specify the AMT tag database to use using the Special_Processing parameters, e.g. AMTDB:MT_Shewanella_ProdTest_Formic_P460.  Define the source jobs by associating with a data package	N	MAA	<JobScript Name="MultiAlign_Aggregator"><Step Number="1" Tool="MultiAlign_Aggregator" Special="ExtractSourceJobFromComment" /></JobScript>	<Param Section="JobParameters" Name="DatasetName" Value="Aggregation" Reqd="Yes" /><Param Section="JobParameters" Name="ParamFileName" Value="" Reqd="Yes" User="Yes" /><Param Section="JobParameters" Name="AlignmentDataset" Value="" Reqd="No" /><Param Section="JobParameters" Name="AMTDB" Value="" Reqd="No" User="Yes" /><Param Section="JobParameters" Name="AMTDBServer" Value="" Reqd="No" User="Yes" /><Param Section="JobParameters" Name="ResultFilename" Value="Aggregation" Reqd="Yes" User="Yes" /><Param Section="JobParameters" Name="MultiAlignSearchType" Value="_isos.csv" Reqd="Yes" User="Yes" /><Param Section="JobParameters" Name="transferFolderPath" Value="" Reqd="Yes" />	\N	1	0	0
49	MSGFPlus	This is the standard script for the "MSGF+" DMS tool	N	MSG	<JobScript Name="MSGFPlus"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSGFPlus"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="IDPicker"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
50	MSGFPlus_DTARefinery	This script uses DTARefinery to refine the DTAs before running "MSGF+"	N	MSG	<JobScript Name="MSGFPlus_DTARefinery"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="DTA_Refinery"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="MSGFPlus"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="IDPicker"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="10" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>	\N	\N	0	0	0
52	MAC_Spectral_Counting	MAC Script for spectral counting	N	SCo	<JobScript Name="MAC_Spectral_Counting"><Step Number="1" Tool="Mage"><Description>Extract DMS results and metadata into cache database</Description></Step><Step Number="2" Tool="APE"><Depends_On Step_Number="1" /><Description>Filters data by iterating over MSGF and PPM\r\nto determine FDRs of 0.1%, 1%, 5%, and 10% based\r\non forward/reverse database searches.</Description></Step><Step Number="3" Tool="Cyclops"><Depends_On Step_Number="2" /><Description>Plots the data and performs a Beta-binomial model analysis on the data to determine significant changes\r\nbetween factors for spectral count results.</Description></Step><Step Number="4" Tool="Results_Cleanup"><Depends_On Step_Number="3" /><Description>Removes superseded Results.db3 files</Description></Step></JobScript>	<Param Section="JobParameters" Name="transferFolderPath" Value="" /><Param Section="JobParameters" Name="AnalysisType" Value="SpectralCounting" Reqd="Yes" /><Param Section="JobParameters" Name="ResultsBaseName" Value="Results" Reqd="Yes" /><Param Section="1_Mage" Name="MageOperations" Value="ExtractFromJobs, GetFactors, GetFDRTables" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="ExtractionSource" Value="JobsFromDataPackageID" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="ExtractionType" Value="Sequest First Hits" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="KeepAllResults" Value="Yes" Reqd="No" Step="1" /><Param Section="1_Mage" Name="ResultFilterSetID" Value="All Pass" Reqd="No" Step="1" /><Param Section="1_Mage" Name="MSGFCutoff" Value="All Pass" Reqd="No" Step="1" /><Param Section="1_Mage" Name="FactorsSource" Value="FactorsFromDataPackageID" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="MageFDRFiles" Value="t_iteration1.txt, t_iteration2_1.txt, t_iteration2_5.txt, t_iteration2_10.txt" Reqd="No" Step="1" /><Param Section="2_Ape" Name="ApeWorkflowName" Value="Sequest_MSGF_MassErr_FDR1_5_10.xml" Reqd="Yes" User="Yes" Step="2" /><Param Section="2_Ape" Name="ApeOperations" Value="RunWorkflow" Reqd="Yes" Step="2" /><Param Section="2_Ape" Name="ApeWorflowStepList" Value="all" Reqd="Yes" Step="2" /><Param Section="2_Ape" Name="ApeCompactDatabase" Value="True" Reqd="Yes" Step="2" /><Param Section="3_Cyclops" Name="CyclopsWorkflowName" Value="ScoOperationsWorkflow.xml" Reqd="Yes" User="Yes" Step="3" /><Param Section="3_Cyclops" Name="Consolidation_Factor" Value="" Reqd="No" User="Yes" Step="3" /><Param Section="3_Cyclops" Name="Fixed_Effect" Value="" Reqd="No" User="Yes" Step="3" />	\N	1	1	0
53	MAC_IMPROV	MAC Script for building IMPROV (Integrated MetaPROteomics Viewer) database	N	IPV	<JobScript Name="MAC_IMPROV"><Step Number="1" Tool="Mage"><Description>Import contents of files from data package to cache database </Description></Step><Step Number="2" Tool="APE"><Depends_On Step_Number="1" /><Description>Import features from MT database into cache database </Description></Step><Step Number="3" Tool="Results_Cleanup"><Depends_On Step_Number="2" /><Description>Removes superseded Results.db3 files</Description></Step></JobScript>	<Param Section="JobParameters" Name="transferFolderPath" Value="" /><Param Section="JobParameters" Name="AnalysisType" Value="Improv" Reqd="Yes" /><Param Section="JobParameters" Name="ResultsBaseName" Value="Results" Reqd="Yes" /><Param Section="1_Mage" Name="MageOperations" Value="ImportDataPackageFiles" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="DataPackageSourceFolderName" Value="ImportFiles" Reqd="Yes" Step="1" /><Param Section="2_Ape" Name="ApeOperations" Value="GetImprovResults,RunWorkflow" Reqd="Yes" Step="2" /><Param Section="2_Ape" Name="ApeMTSServer" Value="" Reqd="Yes" Step="2" User="Yes" /><Param Section="2_Ape" Name="ApeMTSDatabase" Value="" Reqd="Yes" Step="2" User="Yes" /><Param Section="2_Ape" Name="ImprovMinPMTQuality" Value="1.0" Reqd="Yes" Step="2" User="Yes" /><Param Section="2_Ape" Name="ApeWorkflowName" Value="BuildImprovNodesEdges.xml" Reqd="Yes" Step="2" /><Param Section="2_Ape" Name="ApeWorflowStepList" Value="all" Reqd="Yes" Step="2" /><Param Section="2_Ape" Name="ApeCompactDatabase" Value="True" Reqd="Yes" Step="2" />	\N	1	0	0
55	SMAQC_MSMS	This script runs SMAQC to generate QC metrics for LC-MS/MS datasets.  This scripts requires that an existing MASIC and MSGF+ job exist for the dataset.  Use the text SourceJob:1234,SourceJob2:5678 in the job's Special_Processing parameters to define the parent MSGF+ and MASIC jobs.  Alternatively, use a job match spec, e.g.: SourceJob:Auto{Tool = "MSGFPlus" AND (Settings_File Like "IonTrapDefSettings_DeconMSN%.xml" OR Settings_File = "IonTrapDefSettings_MSConvert.xml") AND Param_File = "MSGFPlus_PartTryp_MetOx_20ppmParTol.txt"}, Job2:Auto{Tool = "MASIC_Finnigan"}	Y	SQC	<JobScript Name="SMAQC_MSMS"><Step Number="1" Tool="SMAQC" Special="ExtractSourceJobFromComment" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>			0	0	0
56	MSGFPlus_MzXml	Search mzXML files with "MSGF+" DMS tool.  Appropriate for datasets with high resolution, profile-mode MS/MS spectra.  Also valid for Agilent datasets with .D folders and MS/MS spectra	Y	MSG	<JobScript Name="MSGFPlus_MzXml"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MSGFPlus"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DataExtractor"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="MSGF"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="IDPicker"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
58	MSDeconv_Bruker	Processes Bruker .D folders with MSDeconv	Y	MSA	<JobScript Name="MSDeconv_Bruker"><Step Number="1" Tool="MSXML_Bruker" /><Step Number="2" Tool="MSDeconv"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="2" Enable_Only="1" /></Step><Step Number="4" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>	\N	\N	0	0	0
59	MSGFPlus_Bruker	Search Bruker mzXML files with "MSGF+" DMS tool.  Also works for TIMS datasets from Maxis_01	Y	MSG	<JobScript Name="MSGFPlus_MzXml"><Step Number="1" Tool="MSXML_Bruker" /><Step Number="2" Tool="MSGFPlus"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DataExtractor"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="MSGF"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="4" Enable_Only="1" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="4" /></Step></JobScript>	\N	\N	0	0	0
60	MAC_Label_Free	MAC Script for building Label Free database	N	LBF	<JobScript Name="MAC_Label_Free"><Step Number="1" Tool="Mage"><Description>Import Factors, Data Package from data package to cache database</Description></Step><Step Number="2" Tool="APE"><Depends_On Step_Number="1" /><Description>Import features from MT database into cache database</Description></Step><Step Number="3" Tool="Cyclops"><Depends_On Step_Number="2" /><Description>Plots the data and performs a Beta-binomial model analysis on the data to determine significant changes between factors for spectral count results.</Description></Step><Step Number="4" Tool="Results_Cleanup"><Depends_On Step_Number="3" /><Description>Removes superseded Results.db3 files</Description></Step></JobScript>	<Param Section="JobParameters" Name="transferFolderPath" Value="" /><Param Section="JobParameters" Name="AnalysisType" Value="LabelFree" Reqd="Yes" /><Param Section="JobParameters" Name="ResultsBaseName" Value="Results" Reqd="Yes" /><Param Section="1_Mage" Name="MageOperations" Value="GetFactors,ImportJobList" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="FactorsSource" Value="FactorsFromDataPackageID" Reqd="Yes" Step="1" /><Param Section="2_Ape" Name="ApeOperations" Value="GetViperResults,RunWorkflow" Reqd="Yes" Step="2" /><Param Section="2_Ape" Name="ApeMTSServer" Value="" Reqd="Yes" User="Yes" Step="2" /><Param Section="2_Ape" Name="ApeMTSDatabase" Value="" Reqd="Yes" User="Yes" Step="2" /><Param Section="2_Ape" Name="ApeMDState" Value="2" Reqd="Yes" User="Yes" Step="2" /><Param Section="2_Ape" Name="ApeMDIniFilename" Value="" Reqd="No" User="Yes" Step="" /><Param Section="2_Ape" Name="ApeWorkflowName" Value="NotClustered10perFDR_05UP_Clustered1ppm_SumClassAbu_2.xml" Reqd="Yes" User="Yes" Step="2" /><Param Section="2_Ape" Name="ApeWorflowStepList" Value="all" Reqd="Yes" Step="2" /><Param Section="2_Ape" Name="ApeCompactDatabase" Value="True" Reqd="Yes" Step="2" /><Param Section="3_Cyclops" Name="CyclopsWorkflowName" Value="LBF_Operation_Log2All.xml" Reqd="Yes" User="Yes" Step="3" /><Param Section="3_Cyclops" Name="Consolidation_Factor" Value="" Reqd="No" User="Yes" Step="3" /><Param Section="3_Cyclops" Name="Fixed_Effect" Value="" Reqd="No" User="Yes" Step="3" />	\N	1	0	0
61	Sequest_Extend_IDPicker	This script should be used to process existing Sequest or XTandem results to process them with IDPicker. Retired in 2021.	N	Seq	<JobScript Name="Sequest_Extend_IDPicker"><Step Number="8" Tool="IDPicker"><Depends_On Step_Number="5" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="8" /></Step></JobScript>	\N	\N	0	0	0
62	MAC_iTRAQ	MAC Script for processing iTRAQ data with interference consideration	Y	ITQ	<JobScript Name="MAC_iTRAQ"><Step Number="1" Tool="AScore"><Description>Run AScore on multiple datasets individually and import resultsinto cache database</Description></Step><Step Number="2" Tool="Mage"><Depends_On Step_Number="1" /><Description>Import Factors, Data Package from data package to cache database</Description></Step><Step Number="3" Tool="IDM"><Depends_On Step_Number="2" /><Description>Calculates interference percentage in MS data</Description></Step><Step Number="4" Tool="APE"><Depends_On Step_Number="3" /><Description>Filters iTRAQ results</Description></Step><Step Number="5" Tool="Cyclops"><Depends_On Step_Number="4" /><Description>Plots the data and performs a Beta-binomial model analysis on the data to determine significant changes between factors for spectral count results.</Description></Step><Step Number="6" Tool="Results_Cleanup"><Depends_On Step_Number="5" /><Description>Removes superseded Results.db3 files</Description></Step></JobScript>	<Param Section="JobParameters" Name="transferFolderPath" Value="" /><Param Section="JobParameters" Name="AnalysisType" Value="iTRAQ" Reqd="Yes" /><Param Section="JobParameters" Name="ResultsBaseName" Value="Results" Reqd="Yes" /><Param Section="1_AScore" Name="AScoreParamFilename" Value="" Reqd="No" User="Yes" Step="1" /><Param Section="1_AScore" Name="AScoreSearchType" Value="msgfplus" Reqd="No" Step="1" /><Param Section="1_AScore" Name="ExtractionType" Value="MSGF+ Synopsis All Proteins" Reqd="No" Step="1" /><Param Section="2_Mage" Name="MageOperations" Value="GetFactors, ImportDataPackageFiles, ImportRawFileList, ImportJobList, GetFDRTables, ExtractFromJobs, ImportReporterIons" Reqd="Yes" Step="2" /><Param Section="2_Mage" Name="DataPackageSourceFolderName" Value="ImportFiles" Reqd="Yes" Step="2" /><Param Section="2_Mage" Name="ExtractionSource" Value="JobsFromDataPackageID" Reqd="Yes" Step="2" /><Param Section="2_Mage" Name="ExtractionType" Value="MSGF+ Synopsis All Proteins" Reqd="No" Step="2" /><Param Section="2_Mage" Name="KeepAllResults" Value="Yes" Reqd="No" Step="2" /><Param Section="2_Mage" Name="MSGFcutoff" Value="All Pass" Reqd="No" Step="2" /><Param Section="2_Mage" Name="ResultFilterSetID" Value="All Pass" Reqd="No" Step="2" /><Param Section="2_Mage" Name="FactorsSource" Value="FactorsFromDataPackageID" Reqd="Yes" Step="2" /><Param Section="2_Mage" Name="ReporterIonSource" Value="JobsFromDataPackageIDForTool" Reqd="Yes" Step="2" /><Param Section="2_Mage" Name="MageFDRFiles" Value="t_iteration1.txt, t_iteration2_1.txt" Reqd="No" Step="2" /><Param Section="4_Ape" Name="ApeOperations" Value="RunWorkflow" Reqd="Yes" Step="4" /><Param Section="4_Ape" Name="ApeWorkflowName" Value="MasterWorkflowSyn.xml" Reqd="Yes" User="Yes" Step="4" /><Param Section="4_Ape" Name="ApeWorkflowStepList" Value="msgfplus, 8plex, 1pctFDR, default, no_ascore, no_precursor_filter, keep_nonquant" Reqd="Yes" Step="4" Comment="When using the MAC Job Wizard, the ApeWorkflowStepList parameter will be auto-defined by MapMACJobParameters, utilizing the Experiment_Labelling parameter and the Ape_Workflow_FDR parameter chosen by the user" /><Param Section="4_Ape" Name="ApeCompactDatabase" Value="True" Reqd="Yes" Step="4" /><Param Section="5_Cyclops" Name="CyclopsWorkflowName" Value="ITQ_ExportOperation.xml" Reqd="Yes" User="Yes" Step="5" /><Param Section="5_Cyclops" Name="RunProteinProphet" Value="False" Reqd="Yes" Step="5" /><Param Section="5_Cyclops" Name="Consolidation_Factor" Value="" Reqd="No" User="Yes" Step="5" /><Param Section="5_Cyclops" Name="Fixed_Effect" Value="" Reqd="No" User="Yes" Step="5" />	<obsolete><Param Label="Experiment Labelling" Name="Experiment_Labelling" Value="8plex" Chooser="experimentLabellingPickList" /><Param Label="Ape Workflow FDR" Name="Ape_Workflow_FDR" Value="default" Chooser="apeWorkflowPickList" /></obsolete>	1	1	1
63	Sequest_DTARefinery_Extend_IDPicker	This script should be used to process existing Sequest_DTARefinery results to process them with IDPicker. Retired in 2021.	N	Seq	<JobScript Name="Sequest_DTARefinery_Extend_IDPicker"><Step Number="10" Tool="IDPicker"><Depends_On Step_Number="6" /></Step><Step Number="11" Tool="Results_Transfer"><Depends_On Step_Number="10" /></Step></JobScript>	\N	\N	0	0	0
64	Sequest_No_IDPicker	This is the Sequest script, with MSGF, but without IDPicker. Retired in 2021.	N	Seq	<JobScript Name="Sequest"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="Sequest"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
65	Sequest_DTARefinery_No_IDPicker	This script uses DTARefinery to refine the DTAs before running "Sequest".  It includes MSGF but not IDPicker. Retired in 2021.	N	Seq	<JobScript Name="Sequest"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="DTA_Refinery"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="Sequest"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
66	MSGFPlus_MzXml_Extend_IDPicker	This script should be used to process existing MSGF+_MzXml results to process them with IDPicker	Y	MSG	<JobScript Name="MSGFPlus_MzXml_Extend_IDPicker"><Step Number="7" Tool="IDPicker"><Depends_On Step_Number="4" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>	\N	\N	0	0	0
67	XTandem_No_IDPicker	This is the XTandem script, with MSGF, but without IDPicker	Y	XTM	<JobScript Name="XTandem"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="XTandem"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
68	XTandem_DTARefinery_No_IDPicker	This script uses DTARefinery to refine the DTAs before running "XTandem". It includes MSGF but not IDPicker. Retired in 2021.	N	XTM	<JobScript Name="XTandem"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="DTA_Refinery"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="XTandem"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>			0	0	0
69	MSGFPlus_DTARefinery_No_IDPicker	This script uses DTARefinery to refine the DTAs before running "MSGF+". It includes MSGF but not IDPicker	N	MSG	<JobScript Name="MSGFPlus_DTARefinery_No_IDPicker"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="DTA_Refinery"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="MSGFPlus"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
70	MSGFPlus_No_IDPicker	This is the MSGF+ script, with MSGF, but without IDPicker	N	MSG	<JobScript Name="MSGFPlus"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSGFPlus"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
71	MSGFPlus_MzXml_No_IDPicker	This is the MSGF+ script, with MSGF, but without IDPicker. This version is appropriate for datasets with high resolution, profile-mode MS/MS spectra	Y	MSG	<JobScript Name="MSGFPlus_MzXml"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MSGFPlus"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DataExtractor"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="MSGF"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="4" Enable_Only="1" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="4" /></Step></JobScript>	\N	\N	0	0	0
72	LipidMapSearch	This script uses the LipidTools software to search a pair of Lipid datasets against the LipidMaps database. This scripts requires two existing DeconTools jobs; one for this dataset (the positive-ion dataset) and one for the related negative-ion dataset.  Use a job match spec to define the jobs, e.g.: SourceJob:Auto{Tool = "Decon2LS_V2" AND Param_File = "LTQ_FT_Lipidomics_2012-04-16.xml"}, Job2:Auto{Tool = "Decon2LS_V2" AND Param_File = "LTQ_FT_Lipidomics_2012-04-16.xml" AND Dataset LIKE "$ThisDataset%NEG"}	N	LMS	<JobScript Name="LipidMapSearch"><Step Number="1" Tool="LipidMapSearch" Special="ExtractSourceJobFromComment" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>			0	0	0
73	ProSightPC	This script imports ProSightPC results then quantifies them using DeconToolsWorkflows	N	PSP	<JobScript Name="ProSightPC"><Step Number="1" Tool="DataImport" /><Step Number="2" Tool="ProSight_Quant"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>			0	0	0
74	MSGFPlus_IMS	This script analyzes IMS MS/MS datasets with IonMobilityMsMsConsole.exe, which itself uses MSGF+	N	MSG	<JobScript Name="MSGFPlus_IMS"><Step Number="1" Tool="MSGFPlus_IMS" Special="ExtractSourceJobFromComment" /><Step Number="2" Tool="DataExtractor"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="MSGF"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="IDPicker"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="4" /></Step></JobScript>	\N	\N	0	0	0
75	MAC_2D_LC	MAC Script for 2D LC	N	DLC	<JobScript Name="MAC_2D_LC"><Step Number="1" Tool="Mage"><Description>Extract DMS results and \r\nmetadata into cache database</Description></Step><Step Number="2" Tool="APE"><Depends_On Step_Number="1" /><Description>Filters data.</Description></Step><Step Number="3" Tool="Cyclops"><Depends_On Step_Number="2" /><Description>Plots the data.</Description></Step></JobScript>	<Param Section="JobParameters" Name="transferFolderPath" Value="" /><Param Section="JobParameters" Name="AnalysisType" Value="2DLC" Reqd="Yes" /><Param Section="JobParameters" Name="ResultsBaseName" Value="Results" Reqd="Yes" /><Param Section="1_Mage" Name="MageOperations" Value="ExtractFromJobs, GetFactors" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="ExtractionSource" Value="JobsFromDataPackageID" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="ExtractionType" Value="MSGFDB First Hits" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="KeepAllResults" Value="Yes" Reqd="No" Step="1" /><Param Section="1_Mage" Name="ResultFilterSetID" Value="All Pass" Reqd="No" Step="1" /><Param Section="1_Mage" Name="MSGFCutoff" Value="All Pass" Reqd="No" Step="1" /><Param Section="1_Mage" Name="FactorsSource" Value="FactorsFromDataPackageID" Reqd="Yes" Step="1" /><Param Section="2_Ape" Name="ApeWorkflowName" Value="APE_2DLC.xml" Reqd="Yes" User="Yes" Step="2" /><Param Section="2_Ape" Name="ApeOperations" Value="RunWorkflow" Reqd="Yes" Step="2" /><Param Section="2_Ape" Name="ApeWorflowStepList" Value="all" Reqd="Yes" Step="2" /><Param Section="2_Ape" Name="ApeCompactDatabase" Value="True" Reqd="Yes" Step="2" /><Param Section="3_Cyclops" Name="CyclopsWorkflowName" Value="Cyclops_2DLC.xml" Reqd="Yes" User="Yes" Step="3" /><Param Section="3_Cyclops" Name="RunProteinProphet" Value="False" Reqd="Yes" User="Yes" Step="3" /><Param Section="3_Cyclops" Name="Consolidation_Factor" Value="" Reqd="No" User="Yes" Step="3" /><Param Section="3_Cyclops" Name="Fixed_Effect" Value="" Reqd="No" User="Yes" Step="3" />	\N	1	0	0
76	PRIDE_Converter	Aggregates Peptide_Hit search results (MS-GF+ or X!Tandem) to create a PX_Submission.px file for upload to ProteomeXchange (PRIDE).  The data package directory should have file PX_Submission_Template.px	Y	PPX	<JobScript Name="PRIDE_Converter"><Step Number="1" Tool="PRIDE_Converter" /></JobScript>	<Param Section="JobParameters" Name="PXSubmissionTemplate" Value="PX_Submission_Template.px" Reqd="Yes" User="Yes" /><Param Section="JobParameters" Name="CreateMGFFiles" Value="True" Reqd="Yes" /><Param Section="JobParameters" Name="DisableMyEMSL" Value="False" Reqd="Yes" /><Param Section="JobParameters" Name="IncludeMZidFiles" Value="True" Reqd="Yes" /><Param Section="JobParameters" Name="AssumeInstrumentDataUnpurged" Value="False" Reqd="Yes" /><Param Section="JobParameters" Name="DatasetName" Value="Aggregation" /><Param Section="JobParameters" Name="CacheFolderRootPath" Value="\\\\protoapps\\MassIVE_Staging" />		1	1	0
77	Isobaric_Labeling	Script for processing iTRAQ 4 plex or 8 plex and TMT; superseded by the MAC_iTRAQ script	N	SIL	<JobScript Name="Isobaric_Labeling"><Step Number="1" Tool="Mage"><Description>Import Factors, Data Package from data package to cache database</Description></Step><Step Number="2" Tool="IDM"><Depends_On Step_Number="1" /><Description>Calculates interference percentage in MS data</Description></Step><Step Number="3" Tool="APE"><Depends_On Step_Number="2" /><Description>Filters iTRAQ results</Description></Step><Step Number="4" Tool="Cyclops"><Depends_On Step_Number="3" /><Description>Plots the data and performs a Beta-binomial model analysis on the data to determine significant changes between factors for spectral count results.\r\n</Description></Step><Step Number="5" Tool="Results_Cleanup"><Depends_On Step_Number="4" /><Description>Removes superseded Results.db3 files</Description></Step></JobScript>	<Param Section="JobParameters" Name="transferFolderPath" Value="" /><Param Section="JobParameters" Name="AnalysisType" Value="iTRAQ" Reqd="Yes" /><Param Section="JobParameters" Name="ResultsBaseName" Value="Results" Reqd="Yes" /><Param Section="1_Mage" Name="MageOperations" Value="GetFactors, ImportDataPackageFiles, ImportRawFileList, ImportJobList, GetFDRTables, ExtractFromJobs, ImportReporterIons" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="DataPackageSourceFolderName" Value="ImportFiles" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="ExtractionSource" Value="JobsFromDataPackageID" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="ExtractionType" Value="MSGF+ Synopsis All Proteins" Reqd="No" Step="1" /><Param Section="1_Mage" Name="KeepAllResults" Value="Yes" Reqd="No" Step="1" /><Param Section="1_Mage" Name="MSGFcutoff" Value="All Pass" Reqd="No" Step="1" /><Param Section="1_Mage" Name="ResultFilterSetID" Value="All Pass" Reqd="No" Step="1" /><Param Section="1_Mage" Name="FactorsSource" Value="FactorsFromDataPackageID" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="ReporterIonSource" Value="JobsFromDataPackageIDForTool" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="MageFDRFiles" Value="t_iteration1.txt, t_iteration2_1.txt" Reqd="No" Step="1" /><Param Section="3_Ape" Name="ApeOperations" Value="RunWorkflow" Reqd="Yes" Step="3" /><Param Section="3_Ape" Name="ApeWorkflowName" Value="MasterWorkflowSyn.xml" Reqd="Yes" User="Yes" Step="3" /><Param Section="3_Ape" Name="ApeWorkflowStepList" Value="msgfplus, 4plex, default, no_ascore, no_precursor_filter, keep_nonquant" Reqd="Yes" User="Yes" Step="3" /><Param Section="3_Ape" Name="ApeCompactDatabase" Value="True" Reqd="Yes" Step="3" />	\N	1	0	0
78	Global_Label-Free_AMT_Tag	This script runs MultiAlign in label free mode matching to MTDB (aligns to an AMT tag database).	N	GLF	<JobScript Name="MAC_Global_Label-Free_AMT_Tag"><Step Number="1" Tool="MultiAlign_Aggregator"><Description>Run MultiAlign in AMT Mode </Description></Step></JobScript>	<Param Section="JobParameters" Name="transferFolderPath" Value="" Reqd="Yes" /><Param Section="JobParameters" Name="DatasetName" Value="Aggregation" Reqd="Yes" /><Param Section="JobParameters" Name="ResultsBaseName" Value="Results" Reqd="Yes" /><Param Section="1_MultiAlign_Aggregator" Name="ParamFileName" Value="" Reqd="Yes" /><Param Section="1_MultiAlign_Aggregator" Name="MultiAlignSearchType" Value="_LCMSFeatures.txt" Reqd="Yes" /><Param Section="1_MultiAlign_Aggregator" Name="AlignmentDataset" Value="" Reqd="No" /><Param Section="1_MultiAlign_Aggregator" Name="AMTDB" Value="" Reqd="No" /><Param Section="1_MultiAlign_Aggregator" Name="AMTDBServer" Value="" Reqd="No" /><Param Section="1_MultiAlign_Aggregator" Name="TargetJobFileList" Value="decon2ls_v2:_LCMSFeatures.txt:nocopy,lcmsfeature_finder:_LCMSFeatures.txt:nocopy" Reqd="Yes" />	<obsolete><Param Label="Mass Tag Database" Name="Mass_Tag_Db" Value="" Chooser="amtDBPicklist" /></obsolete>	1	1	0
79	Isobaric_Labeling_No_IDM	Script for processing iTRAQ 4 plex or 8 plex and TMT without IDM step	N	SIL	<JobScript Name="MAC_iTRAQ"><Step Number="1" Tool="Mage"><Description>Import Factors, Data Package from data package to cache database</Description></Step><Step Number="2" Tool="APE"><Depends_On Step_Number="1" /><Description>Filters iTRAQ results</Description></Step><Step Number="3" Tool="Results_Cleanup"><Depends_On Step_Number="2" /><Description>Removes superseded Results.db3 files</Description></Step></JobScript>	<Param Section="JobParameters" Name="transferFolderPath" Value="" /><Param Section="JobParameters" Name="AnalysisType" Value="iTRAQ" Reqd="Yes" /><Param Section="JobParameters" Name="ResultsBaseName" Value="Results" Reqd="Yes" /><Param Section="1_Mage" Name="MageOperations" Value="GetFactors, ImportDataPackageFiles, ImportRawFileList, ImportJobList, GetFDRTables, ExtractFromJobs, ImportReporterIons" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="DataPackageSourceFolderName" Value="ImportFiles" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="ExtractionSource" Value="JobsFromDataPackageID" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="ExtractionType" Value="Sequest First Hits" Reqd="No" Step="1" /><Param Section="1_Mage" Name="KeepAllResults" Value="Yes" Reqd="No" Step="1" /><Param Section="1_Mage" Name="MSGFcutoff" Value="All Pass" Reqd="No" Step="1" /><Param Section="1_Mage" Name="ResultFilterSetID" Value="All Pass" Reqd="No" Step="1" /><Param Section="1_Mage" Name="FactorsSource" Value="FactorsFromDataPackageID" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="ReporterIonSource" Value="JobsFromDataPackageIDForTool" Reqd="Yes" Step="1" /><Param Section="1_Mage" Name="MageFDRFiles" Value="t_iteration1.txt, t_iteration2_1.txt" Reqd="No" Step="1" /><Param Section="3_Ape" Name="ApeOperations" Value="RunWorkflow" Reqd="Yes" Step="2" /><Param Section="3_Ape" Name="ApeWorkflowName" Value="MasterWorkflow.xml" Reqd="Yes" User="Yes" Step="2" /><Param Section="3_Ape" Name="ApeWorflowStepList" Value="all" Reqd="Yes" Step="2" /><Param Section="3_Ape" Name="ApeCompactDatabase" Value="True" Reqd="Yes" Step="2" />	\N	1	0	0
81	MSGFPlus_SplitFasta	This script runs MSGF+ against a large fasta file that has been split into multiple parts.  It spawns several parallel copies of MSGF+, which each search a separate part of the Fasta file.  Note that the IDPicker step is included so that the PepXML file will be created.  However, IDPicker will not actually be run since idpQonvert cannot handle the large Fasta files that are typically used with this script.	N	MSG	<JobScript Name="MSGFPlus_SplitFasta"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSGFPlus" Special="Clone"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractorSplitFasta"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="IDPicker"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
82	Decon2LS_V2_MzXML	This script runs the DeconToolsAutoProcessor, processing a mzXML file. Following Decon2LS, the LCMSFeatureFinder processes the Decon2LS results.  The Settings file must contain an LCMSFeatureFinder section (with LCMSFeatureFinderIniFile defining the Feature Finder .Ini file to use).  However, LCMSFeatureFinder will be skipped if that section is empty.	Y	DLS	<JobScript Name="Decon2LS_V2_MzXML"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="Decon2LS_V2"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="LCMSFeatureFinder"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="3" Enable_Only="1" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="3" /></Step></JobScript>	\N	\N	0	0	0
84	PeptideAtlas	Aggregates Peptide_Hit search results (Sequest, X!Tandem, or MSGF) to create uploadable package for Peptide Atlas; retired in 2020	N	PAR	<JobScript Name="PeptideAtlas"><Step Number="1" Tool="RepoPkgr" /></JobScript>	<Param Section="Submission_Info" Name="submitter" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="datasetTag" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="datasetTitle" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="publicReleaseDate" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="Contributors" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="Submitter Organization" Value="Pacific Northwest National Laboratory" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="Lab head" Value="Richard D. Smith" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="Lab head e-mail" Value="dick.smith@pnnl.gov" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="Lab head organization" Value="Biological Sciences Division" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="Lab head country" Value="United States" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="publication" Reqd="Yes" User="Yes" /><Param Section="Submission_Info" Name="summary" Reqd="No" User="Yes" /><Param Section="Submission_Info" Name="growth protocol" Reqd="No" User="Yes" /><Param Section="Submission_Info" Name="treatment protocol" Reqd="No" User="Yes" /><Param Section="Submission_Info" Name="extraction protocol" Reqd="No" User="Yes" /><Param Section="Submission_Info" Name="separation protocol" Reqd="No" User="Yes" /><Param Section="Submission_Info" Name="digestion protocol" Reqd="No" User="Yes" /><Param Section="Submission_Info" Name="acquisition protocol" Reqd="No" User="Yes" /><Param Section="Submission_Info" Name="informatics protocol" Reqd="No" User="Yes" /><Param Section="RepoAccount" Name="RepoAccountServer" Reqd="No" User="Yes" /><Param Section="RepoAccount" Name="RepoAccountUser" Reqd="No" User="Yes" /><Param Section="RepoAccount" Name="RepoAccountPassword" Reqd="No" User="Yes" /><Param Section="PackageContents" Name="IncludeMSGFResults" Value="True" Reqd="Yes" /><Param Section="PackageContents" Name="IncludeSequestResults" Value="False" Reqd="Yes" /><Param Section="PackageContents" Name="IncludeInstrumentData" Value="True" Reqd="Yes" /><Param Section="PackageContents" Name="IncludeMzXMLFiles" Value="True" Reqd="Yes" /><Param Section="JobParameters" Name="Repository" Value="PeptideAtlas" /><Param Section="JobParameters" Name="DatasetName" Value="Aggregation" /><Param Section="JobParameters" Name="CacheFolderRootPath" Value="\\\\protoapps\\PeptideAtlas_Staging" />		1	0	0
85	MSGFPlus_HPC	This is the standard script for running MSGF+ on HPC (deception.pnl.gov)	N	MSG	<JobScript Name="MSGFPlus_HPC"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSGFPlus_HPC"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="IDPicker"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
86	MODa	This is the standard script for the MODa DMS tool.  The MSGF step is required to summarize the MODa results, creating stats for http://dms2.pnl.gov/analysis_job_psm/report/-/-/MODa	Y	MODa	<JobScript Name="MODa"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MODa"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="IDPicker"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
87	MODa_DTARefinery	This script uses DTARefinery to refine the DTAs before running MODa.  The MSGF step is required to summarize the MODa results, creating stats for http://dms2.pnl.gov/analysis_job_psm/report/-/-/MODa	N	MODa	<JobScript Name="MODa_DTARefinery"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="DTA_Refinery"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="MODa"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="IDPicker"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="10" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>			0	0	0
88	GlyQ-IQ	This script runs GlyQ-IQ to identify glycans	N	GLY	<JobScript Name="Decon2LS_V2"><Step Number="1" Tool="DeconPeakDetector" /><Step Number="2" Tool="GlyQ_IQ"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>			0	0	0
89	MSPathFinder	This is the standard script for the MSPathFinder DMS tool (primarily used for top down).  The MSGF step is required to summarize the MSPathFinder results, creating stats for http://dms2.pnl.gov/analysis_job_psm/report/-/-/MSPathFinder	Y	MSP	<JobScript Name="MSPathFinder"><Step Number="1" Tool="PBF_Gen" /><Step Number="2" Tool="ProMex"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="MSPathFinder"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
90	MSGFPlus_MzML	Search mzML files with MS-GF+.  Intended for analyzing all of our supported instrument types (including Thermo).  The MzML file will typically have centroided MS1 and MS2 spectra (though that option is defined by the settings file).  Runs MzRefinery if the settings file has an entry for MzRefParamFile.	Y	MSG	<JobScript Name="MSGFPlus_MzML"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="Mz_Refinery"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="MSGFPlus"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><!-- Note that MSGF won't actually be run, but the _syn_MSGF.txt and _fht_MSGF.txt files \r\nwill be created (used by Mage) and the PMS Stats will be computed \r\nand stored in T_Analysis_Job_PSM_Stats --><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="IDPicker"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
91	MSGFPlus_MzML_NoRefine	Search MzML files with "MSGF+" DMS tool.  Intended for analyzing all of our supported instrument types (including Thermo).  The MzML file will typically have centroided MS1 and MS2 spectra (though that option is defined by the settings file).  Does not run MzRefinery	Y	MSG	<JobScript Name="MSGFPlus_MzML_NoRefine"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MSGFPlus"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DataExtractor"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="MSGF"><!-- Note that MSGF won't actually be run, but the _syn_MSGF.txt and _fht_MSGF.txt files \r\nwill be created (used by Mage) and the PMS Stats will be computed \r\nand stored in T_Analysis_Job_PSM_Stats --><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="IDPicker"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
92	MAC_TMT10Plex	MAC Script for processing TMT data with interference consideration; defaults to 10-plex TMT, but also supports 11, 16, and 18-plex TMT	Y	ITQ	<JobScript Name="MAC_iTRAQ"><Step Number="1" Tool="AScore"><Description>Run AScore on multiple datasets individually and import resultsinto cache database</Description></Step><Step Number="2" Tool="Mage"><Depends_On Step_Number="1" /><Description>Import Factors, Data Package from data package to cache database</Description></Step><Step Number="3" Tool="IDM"><Depends_On Step_Number="2" /><Description>Calculates interference percentage in MS data</Description></Step><Step Number="4" Tool="APE"><Depends_On Step_Number="3" /><Description>Filters iTRAQ results</Description></Step><Step Number="5" Tool="Cyclops"><Depends_On Step_Number="4" /><Description>Plots the data and performs a Beta-binomial model analysis on the data to determine significant changes between factors for spectral count results.</Description></Step><Step Number="6" Tool="Results_Cleanup"><Depends_On Step_Number="5" /><Description>Removes superseded Results.db3 files</Description></Step></JobScript>	<Param Section="JobParameters" Name="transferFolderPath" Value="" /><Param Section="JobParameters" Name="AnalysisType" Value="iTRAQ" Reqd="Yes" /><Param Section="JobParameters" Name="ResultsBaseName" Value="Results" Reqd="Yes" /><Param Section="1_AScore" Name="AScoreParamFilename" Value="" Reqd="No" User="Yes" Step="1" /><Param Section="1_AScore" Name="AScoreSearchType" Value="msgfplus" Reqd="No" Step="1" /><Param Section="1_AScore" Name="ExtractionType" Value="MSGF+ Synopsis All Proteins" Reqd="No" Step="1" /><Param Section="2_Mage" Name="MageOperations" Value="GetFactors, ImportDataPackageFiles, ImportRawFileList, ImportJobList, GetFDRTables, ExtractFromJobs, ImportReporterIons" Reqd="Yes" Step="2" /><Param Section="2_Mage" Name="DataPackageSourceFolderName" Value="ImportFiles" Reqd="Yes" Step="2" /><Param Section="2_Mage" Name="ExtractionSource" Value="JobsFromDataPackageID" Reqd="Yes" Step="2" /><Param Section="2_Mage" Name="ExtractionType" Value="MSGF+ Synopsis All Proteins" Reqd="No" Step="2" /><Param Section="2_Mage" Name="KeepAllResults" Value="Yes" Reqd="No" Step="2" /><Param Section="2_Mage" Name="MSGFcutoff" Value="All Pass" Reqd="No" Step="2" /><Param Section="2_Mage" Name="ResultFilterSetID" Value="All Pass" Reqd="No" Step="2" /><Param Section="2_Mage" Name="FactorsSource" Value="FactorsFromDataPackageID" Reqd="Yes" Step="2" /><Param Section="2_Mage" Name="ReporterIonSource" Value="JobsFromDataPackageIDForTool" Reqd="Yes" Step="2" /><Param Section="2_Mage" Name="MageFDRFiles" Value="t_iteration1.txt, t_iteration2_1.txt" Reqd="No" Step="2" /><Param Section="4_Ape" Name="ApeOperations" Value="RunWorkflow" Reqd="Yes" Step="4" /><Param Section="4_Ape" Name="ApeWorkflowName" Value="MasterWorkflowSyn.xml" Reqd="Yes" User="Yes" Step="4" /><Param Section="4_Ape" Name="ApeWorkflowStepList" Value="msgfplus, TMT10Plex, 1pctFDR, default, no_ascore, no_precursor_filter, keep_nonquant" Reqd="Yes" Step="4" Comment="When using the MAC Job Wizard, the ApeWorkflowStepList parameter will be auto-defined by MapMACJobParameters, utilizing the Experiment_Labelling parameter and the Ape_Workflow_FDR parameter chosen by the user" /><Param Section="4_Ape" Name="ApeCompactDatabase" Value="True" Reqd="Yes" Step="4" /><Param Section="5_Cyclops" Name="CyclopsWorkflowName" Value="ITQ_ExportOperation.xml" Reqd="Yes" User="Yes" Step="5" /><Param Section="5_Cyclops" Name="RunProteinProphet" Value="False" Reqd="Yes" Step="5" /><Param Section="5_Cyclops" Name="Consolidation_Factor" Value="" Reqd="No" User="Yes" Step="5" /><Param Section="5_Cyclops" Name="Fixed_Effect" Value="" Reqd="No" User="Yes" Step="5" />	<obsolete><Param Label="Experiment Labelling" Name="Experiment_Labelling" Value="8plex" Chooser="experimentLabellingPickList" /><Param Label="Ape Workflow FDR" Name="Ape_Workflow_FDR" Value="default" Chooser="apeWorkflowPickList" /></obsolete>	1	1	1
93	ProMex	This script creates a PBF file and then deisotopes the data using ProMex (intended for top down datasets).	Y	PMX	<JobScript Name="ProMex"><Step Number="1" Tool="PBF_Gen" /><Step Number="2" Tool="ProMex" Special="Job_Results"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="2" Enable_Only="1" /></Step><Step Number="4" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>			0	0	0
94	ProMex_Bruker	This script converts a Bruker .D folder to a mzML file then deisotopes the data using ProMex (intended for top down datasets).	Y	PMX	<JobScript Name="ProMex_Bruker"><Step Number="1" Tool="MSXML_Bruker" /><Step Number="2" Tool="ProMex" Special="Job_Results"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="2" Enable_Only="1" /></Step><Step Number="4" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>			0	0	0
95	NOMSI	This script processes Bruker .D folders to export spectra as XML files then process them with NOMSI	Y	NOM	<JobScript Name="NOMSI"><Step Number="1" Tool="Bruker_DA_Export" /><Step Number="2" Tool="NOMSI"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>			0	0	0
96	MODPlus	This is the standard script for the MODPlus DMS tool.  Runs MzRefinery if the settings file has an entry for MzRefParamFile.  The MSGF step is required to summarize the MODPlus results, creating stats for http://dms2.pnl.gov/analysis_job_psm/report/-/-/MODPlus	Y	MODPlus	<JobScript Name="MODPlus"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="Mz_Refinery"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="MODPlus"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="IDPicker"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
97	MODPlus_NoRefine	This is the standard script for the MODPlus DMS tool.  Runs MzRefinery if the settings file has an entry for MzRefParamFile.  Does not run MzRefinery.  The MSGF step is required to summarize the MODPlus results, creating stats for http://dms2.pnl.gov/analysis_job_psm/report/-/-/MODPlus	Y	MODPlus	<JobScript Name="MODPlus_NoRefine"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MODPlus"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DataExtractor"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="MSGF"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="IDPicker"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
98	QC-ART	This script runs the QC-ART tool to produce quality control metrics using MASIC and SMAQC results (SMAQC requires MSGF+ results). QC-ART was developed by Lisa Bramer, Allison Thompson, Bryan Stanfill, and Bobbie-Joe Webb-Robertson.  This scripts requires that an existing SMAQC job and an existing MASIC job exist for the dataset.  Use the text SourceJob:120001,Job2:120002 in the job's Special_Processing parameters to define the parent jobs.  Alternatively, use a job match spec, e.g.: SourceJob:Auto{Tool = "SMAQC_MSMS"}, Job2:Auto{Tool = "MASIC_Finnigan"}	Y	QCA	<JobScript Name="QC-ART"><Step Number="1" Tool="QC_ART" Special="ExtractSourceJobFromComment" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>	\N	\N	0	0	0
99	PBF_Gen	Creates a .pbf file for the given dataset.	Y	PBF	<JobScript Name="PBF_Gen"><Step Number="1" Tool="PBF_Gen" /><Step Number="2" Tool="Results_Transfer"><Depends_On Step_Number="1" /></Step></JobScript>	\N	\N	0	0	0
100	NOMSI_mzXML	This script processes Bruker .D folders to export spectra as mzXML files then process them with NOMSI	N	NOM	<JobScript Name="NOMSI"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="NOMSI"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>			0	0	0
101	MSGFPlus_DTARefinery_SplitFasta	This script uses DTARefinery to refine the DTAs before running "MSGF+".  It then runs MSGF+ against a large fasta file that has been split into multiple parts. It spawns several parallel copies of MSGF+, which each search a separate part of the Fasta file.  Note that the IDPicker step is included so that the PepXML file will be created.  However, IDPicker will not actually be run since idpQonvert cannot handle the large Fasta files that are typically used with this script.	N	MSG	<JobScript Name="MSGFPlus_SplitFasta"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="DTA_Refinery"><Depends_On Step_Number="1" Test="No_Parameters" /></Step><Step Number="3" Tool="MSMSSpectraPreprocessor"><Depends_On Step_Number="2" Test="No_Parameters" /></Step><Step Number="4" Tool="MSGFPlus" Special="Clone"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractorSplitFasta"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="IDPicker"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="10" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>	\N	\N	0	0	0
102	MSGFPlus_MzML_SplitFasta	This script uses MZRefinery to refine the DTAs before running "MSGF+" (parameter MzRefParamFile controls MzRefinery settings).  It then runs MSGF+ against a large fasta file that has been split into multiple parts. It spawns several parallel copies of MSGF+, which each search a separate part of the Fasta file.  Note that the IDPicker step is included so that the PepXML file will be created.  However, IDPicker will not actually be run since idpQonvert cannot handle the large Fasta files that are typically used with this script.	Y	MSG	<JobScript Name="MSGFPlus_MzML_SplitFasta"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="Mz_Refinery"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="MSGFPlus" Special="Clone"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractorSplitFasta"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSGF"><!-- Note that MSGF won't actually be run, but the _syn_MSGF.txt and _fht_MSGF.txt files \r\nwill be created (used by Mage) and the PMS Stats will be computed \r\nand stored in T_Analysis_Job_PSM_Stats --><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="IDPicker"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="6" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="6" /></Step></JobScript>	\N	\N	0	0	0
103	Formularity_Bruker	This script processes Bruker .D folders to export spectra as XML files then process them with Formularity	N	FRM	<JobScript Name="Formularity_Bruker"><Step Number="1" Tool="Bruker_DA_Export" /><Step Number="2" Tool="Formularity"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>			0	0	0
104	TopPIC	This is the standard script for the "TopPIC" DMS tool	Y	TPC	<JobScript Name="TopPIC"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="TopFD"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="TopPIC"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSAlign_Quant"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>			0	0	0
105	Formularity_Thermo	This script processes Thermo .raw files to export spectra as TSV files (using ThermoPeakDataExporter) then process them with Formularity	Y	FRM	<JobScript Name="Formularity_Thermo"><Step Number="1" Tool="ThermoPeakDataExporter" /><Step Number="2" Tool="Formularity"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>			0	0	0
106	MSGFPlus_DeconMSn_MzRefinery	Create a _dta.txt file using DeconMSn, then convert that file to mzML and run MZRefinery.  Next search the MzML file with MSGF+.  Note: for typical MzRefinery searches, use pipeline script MSGFPlus_MzML	N	MSG	<JobScript Name="MSGFPlus_DeconMSn_MzRefinery"><Step Number="1" Tool="DTA_Gen" /><Step Number="2" Tool="MSXML_Gen" /><Step Number="3" Tool="Mz_Refinery"><Depends_On Step_Number="1" /><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="MSGFPlus"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="DataExtractor"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="MSGF"><!-- Note that MSGF won't actually be run, but the _syn_MSGF.txt and _fht_MSGF.txt files \r\nwill be created (used by Mage) and the PMS Stats will be computed \r\nand stored in T_Analysis_Job_PSM_Stats --><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="IDPicker"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="2" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="10" Tool="Results_Transfer"><Depends_On Step_Number="3" Test="Target_Skipped" /><Depends_On Step_Number="7" Enable_Only="1" /></Step><Step Number="11" Tool="Results_Transfer"><Depends_On Step_Number="7" /></Step></JobScript>			0	0	0
107	Formularity_Bruker_Decon	This script processes Bruker .D folders to export spectra as DeconTools _peaks.txt files, then process the spectra with Formularity	Y	FRM	<JobScript Name="Formularity_Bruker"><Step Number="1" Tool="Decon2LS_V2" /><Step Number="2" Tool="Formularity"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="Results_Transfer"><Depends_On Step_Number="2" /></Step></JobScript>			0	0	0
109	MSGFPlus_MzML_SplitFasta_NoRefine	Search mzML files with MS-GF+. Will split a large fasta file into multiple parts. Spawns several parallel copies of MSGF+, which each search a separate part of the Fasta file.  Note that the IDPicker step is included so that the PepXML file will be created.  However, IDPicker will not actually be run since idpQonvert cannot handle the large Fasta files that are typically used with this script.	Y	MSG	<JobScript Name="MSGFPlus_MzML_SplitFasta"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MSGFPlus" Special="Clone"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DataExtractorSplitFasta"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="MSGF"><!-- Note that MSGF won't actually be run, but the _syn_MSGF.txt and _fht_MSGF.txt files \r\nwill be created (used by Mage) and the PMS Stats will be computed \r\nand stored in T_Analysis_Job_PSM_Stats --><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="IDPicker"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>			0	0	0
110	MaxQuant	Script for running MaxQuant against a single dataset (supports both native mass spec files and .mzML files)	Y	MXQ	<JobScript Name="MaxQuant"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MaxqPeak"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="MaxqS1"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="MaxqS2"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MaxqS3"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Cleanup"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="DataExtractor"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="IDPicker"><Depends_On Step_Number="7" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="8" Enable_Only="1" /></Step><Step Number="10" Tool="Results_Transfer"><Depends_On Step_Number="8" /></Step></JobScript>			0	0	0
111	MaxQuant_DataPkg	Script for running MaxQuant against datasets in a data package	Y	MXQ	<JobScript Name="MaxQuant"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MaxqPeak"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="MaxqS1"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="MaxqS2"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MaxqS3"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Cleanup"><Depends_On Step_Number="5" /></Step><Step Number="7" Tool="DataExtractor"><Depends_On Step_Number="6" /></Step><Step Number="8" Tool="IDPicker"><Depends_On Step_Number="7" /></Step><Step Number="9" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="8" Enable_Only="1" /></Step><Step Number="10" Tool="Results_Transfer"><Depends_On Step_Number="8" /></Step></JobScript>	<Param Section="JobParameters" Name="CreateMzMLFiles" Value="False" Reqd="Yes" User="Yes" /><Param Section="JobParameters" Name="DatasetName" Value="Aggregation" Reqd="Yes" /><Param Section="JobParameters" Name="CacheFolderRootPath" Value="\\\\proto-9\\MaxQuant_Staging" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="MSXMLGenerator" Value="MSConvert.exe" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="MSXMLOutputType" Value="mzML" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="CentroidMSXML" Value="True" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="CentroidPeakCountToRetain" Value="-1" Reqd="Yes" /><Param Section="PeptideSearch" Name="ParamFileName" Value="MaxQuant_Tryp_Dyn_MetOx_NTermAcet_20ppmParTol.xml" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ParamFileStoragePath" Value="\\\\gigasax\\DMS_Parameter_Files\\MaxQuant" Reqd="Yes" /><Param Section="PeptideSearch" Name="OrganismName" Value="Homo_sapiens" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ProteinCollectionList" Value="" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ProteinOptions" Value="seq_direction=forward,filetype=fasta" Reqd="Yes" /><Param Section="PeptideSearch" Name="LegacyFastaFileName" Value="na" Reqd="Yes" />		1	1	0
113	DiaNN	Script for running DIA-NN against a single dataset	Y	DNN	<JobScript Name="DiaNN"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="DiaNN_SpecLib"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DiaNN"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="4" Enable_Only="1" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="4" /></Step></JobScript>			0	0	0
108	MSFragger	Script for running MSFragger against a single dataset	N	MSF	<JobScript Name="MSFragger"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MSFragger"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="PepProtProphet"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="4" Enable_Only="1" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="4" /></Step></JobScript>			0	0	0
114	DiaNN_DataPkg	Script for running DIA-NN against datasets in a data package	Y	DNN	<JobScript Name="DiaNN_DataPkg"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="DiaNN_SpecLib"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DiaNN"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="4" Enable_Only="1" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="4" /></Step></JobScript>	<Param Section="JobParameters" Name="CreateMzMLFiles" Value="False" Reqd="Yes" User="Yes" /><Param Section="JobParameters" Name="DatasetName" Value="Aggregation" Reqd="Yes" /><Param Section="JobParameters" Name="CacheFolderRootPath" Value="\\\\proto-9\\DiaNN_Staging" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="MSXMLGenerator" Value="MSConvert.exe" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="MSXMLOutputType" Value="mzML" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="CentroidMSXML" Value="True" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="CentroidPeakCountToRetain" Value="-1" Reqd="Yes" /><Param Section="PeptideSearch" Name="ParamFileName" Value="DiaNN_Tryp_Dyn_MetOx_Stat_CysAlk.txt" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ParamFileStoragePath" Value="\\\\gigasax\\DMS_Parameter_Files\\DiaNN" Reqd="Yes" /><Param Section="PeptideSearch" Name="OrganismName" Value="Homo_sapiens" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ProteinCollectionList" Value="" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ProteinOptions" Value="seq_direction=forward,filetype=fasta" Reqd="Yes" /><Param Section="PeptideSearch" Name="LegacyFastaFileName" Value="na" Reqd="Yes" />		1	1	0
115	FragPipe	Script for using FragPipe to run MSFragger against a single dataset	Y	MSF	<JobScript Name="FragPipe"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="FragPipe"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DataExtractor"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="3" Enable_Only="1" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="3" /></Step></JobScript>	\N	\N	0	0	0
116	FragPipe_DataPkg	Script for using FragPipe to run MSFragger against datasets in a data package	Y	MSF	<JobScript Name="FragPipe"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="FragPipe"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DataExtractor"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="3" Enable_Only="1" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="3" /></Step></JobScript>	<Param Section="JobParameters" Name="CreateMzMLFiles" Value="False" Reqd="Yes" User="Yes" /><Param Section="JobParameters" Name="DatasetName" Value="Aggregation" Reqd="Yes" /><Param Section="JobParameters" Name="CacheFolderRootPath" Value="\\\\proto-9\\MSFragger_Staging" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="MSXMLGenerator" Value="MSConvert.exe" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="MSXMLOutputType" Value="mzML" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="CentroidMSXML" Value="True" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="CentroidPeakCountToRetain" Value="-1" Reqd="Yes" /><Param Section="PeptideSearch" Name="ParamFileName" Value="FragPipe_LFQ-MBR_Tryp_Dyn_MetOx_ProtNTermAcet_20ppmParTol.workflow" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ParamFileStoragePath" Value="\\\\gigasax\\DMS_Parameter_Files\\FragPipe" Reqd="Yes" /><Param Section="PeptideSearch" Name="OrganismName" Value="Homo_sapiens" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ProteinCollectionList" Value="" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ProteinOptions" Value="seq_direction=decoy,filetype=fasta" Reqd="Yes" /><Param Section="PeptideSearch" Name="LegacyFastaFileName" Value="na" Reqd="Yes" /><Param Section="FragPipe" Name="AutoDefineExperimentGroupWithDatasetName" Value="true" Reqd="No" /><Param Section="FragPipe" Name="AutoDefineExperimentGroupWithExperimentName" Value="false" Reqd="No" /><Param Section="FragPipe" Name="DatabaseSplitCount" Value="1" comment="If greater than one, split the FASTA file into the specified number of parts" Reqd="No" /><Param Section="FragPipe" Name="FragPipeMemorySizeMB" Value="49152" Reqd="No" /><Param Section="FragPipe" Name="ReporterIonMode" Value="auto" comment="This can be used to specify tmt6 or tmt10 instead of tmt11, or to specify tmt18 instead of tmt16" Reqd="No" />	\N	1	1	0
117	DiaNN_timsTOF	Script for running DIA-NN against a single timsTOF dataset. The instrument data is not converted to .mzML; DIA-NN directly reads the .tdf file	Y	DNN	<JobScript Name="DiaNN_timsTOF"><Step Number="1" Tool="DiaNN_SpecLib"></Step><Step Number="2" Tool="DiaNN"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DataExtractor"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="Results_Transfer"><Depends_On Step_Number="3" /></Step></JobScript>	\N	\N	0	0	0
118	DiaNN_timsTOF_DataPkg	Script for running DIA-NN against timsTOF datasets in a data package. The instrument data is not converted to .mzML; DIA-NN directly reads the .tdf files	Y	DNN	<JobScript Name="DiaNN_timsTOF_DataPkg"><Step Number="1" Tool="DiaNN_SpecLib"></Step><Step Number="2" Tool="DiaNN"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="DataExtractor"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="Results_Transfer"><Depends_On Step_Number="3" /></Step></JobScript>	<Param Section="JobParameters" Name="CreateMzMLFiles" Value="False" Reqd="Yes" User="Yes" /><Param Section="JobParameters" Name="DatasetName" Value="Aggregation" Reqd="Yes" /><Param Section="JobParameters" Name="CacheFolderRootPath" Value="\\\\proto-9\\DiaNN_Staging" Reqd="Yes" /><Param Section="PeptideSearch" Name="ParamFileName" Value="DiaNN_Tryp_Dyn_MetOx_Stat_CysAlk.txt" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ParamFileStoragePath" Value="\\\\gigasax\\DMS_Parameter_Files\\DiaNN" Reqd="Yes" /><Param Section="PeptideSearch" Name="OrganismName" Value="Homo_sapiens" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ProteinCollectionList" Value="" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ProteinOptions" Value="seq_direction=forward,filetype=fasta" Reqd="Yes" /><Param Section="PeptideSearch" Name="LegacyFastaFileName" Value="na" Reqd="Yes" />	\N	1	1	0
51	MSAlign	This is the standard script for the "MSAlign" DMS tool.  Use this script with Thermo instruments	Y	MSA	<JobScript Name="MSAlign"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MSDeconv"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="MSAlign"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSAlign_Quant"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
57	MSAlign_Bruker	Processes Bruker .D folders with MSAlign	N	MSA	<JobScript Name="MSAlign_Bruker"><Step Number="1" Tool="MSXML_Bruker" /><Step Number="2" Tool="MSDeconv"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="MSAlign"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSAlign_Quant"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>	\N	\N	0	0	0
80	MSAlign_Histone	This is the standard script for the "MSAlign_Histone" DMS tool.  Use this script with Thermo instruments	N	MSA	<JobScript Name="MSAlign_Histone"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MSDeconv"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="MSAlign_Histone"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="MSAlign_Quant"><Depends_On Step_Number="4" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="5" Enable_Only="1" /></Step><Step Number="7" Tool="Results_Transfer"><Depends_On Step_Number="5" /></Step></JobScript>			0	0	0
112	MSFragger_DataPkg	Script for running MSFragger against datasets in a data package	N	MSF	<JobScript Name="MSFragger"><Step Number="1" Tool="MSXML_Gen" /><Step Number="2" Tool="MSFragger"><Depends_On Step_Number="1" /></Step><Step Number="3" Tool="PepProtProphet"><Depends_On Step_Number="2" /></Step><Step Number="4" Tool="DataExtractor"><Depends_On Step_Number="3" /></Step><Step Number="5" Tool="Results_Transfer"><Depends_On Step_Number="1" Test="Target_Skipped" /><Depends_On Step_Number="4" Enable_Only="1" /></Step><Step Number="6" Tool="Results_Transfer"><Depends_On Step_Number="4" /></Step></JobScript>	<Param Section="JobParameters" Name="CreateMzMLFiles" Value="False" Reqd="Yes" User="Yes" /><Param Section="JobParameters" Name="DatasetName" Value="Aggregation" Reqd="Yes" /><Param Section="JobParameters" Name="CacheFolderRootPath" Value="\\\\proto-9\\MSFragger_Staging" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="MSXMLGenerator" Value="MSConvert.exe" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="MSXMLOutputType" Value="mzML" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="CentroidMSXML" Value="True" Reqd="Yes" /><Param Section="MSXMLGenerator" Name="CentroidPeakCountToRetain" Value="-1" Reqd="Yes" /><Param Section="PeptideSearch" Name="ParamFileName" Value="MSFragger_Tryp_Dyn_MetOx_ProtNTermAcet_20ppmParTol.params" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ParamFileStoragePath" Value="\\\\gigasax\\DMS_Parameter_Files\\MSFragger" Reqd="Yes" /><Param Section="PeptideSearch" Name="OrganismName" Value="Homo_sapiens" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ProteinCollectionList" Value="" Reqd="Yes" User="Yes" /><Param Section="PeptideSearch" Name="ProteinOptions" Value="seq_direction=decoy,filetype=fasta" Reqd="Yes" /><Param Section="PeptideSearch" Name="LegacyFastaFileName" Value="na" Reqd="Yes" /><Param Section="MSFragger" Name="MSFraggerJavaMemorySize" Value="49152" Reqd="No" /><Param Section="Philosopher" Name="MatchBetweenRuns" Value="true" Reqd="No" /><Param Section="Philosopher" Name="RunPeptideProphet" Value="auto" Reqd="No" /><Param Section="Philosopher" Name="RunProteinProphet" Value="auto" Reqd="No" /><Param Section="Philosopher" Name="RunPercolator" Value="auto" Reqd="No" /><Param Section="Philosopher" Name="GeneratePeptideLevelSummary" Value="auto" comment="Runs iProphet; only used if multiple datasets and they have different experiment group names" Reqd="No" /><Param Section="Philosopher" Name="GenerateProteinLevelSummary" Value="auto" comment="Runs Abacus; only used if multiple datasets and they have different experiment group names" Reqd="No" /><Param Section="Philosopher" Name="MS1QuantDisabled" Value="false" comment="If true, disables FreeQuant and IonQuant" Reqd="No" /><Param Section="Philosopher" Name="RunFreeQuant" Value="auto" Reqd="No" /><Param Section="Philosopher" Name="RunIonQuant" Value="auto" Reqd="No" /><Param Section="Philosopher" Name="ReporterIonMode" Value="auto" comment="This can be used to specify tmt6 or tmt10 instead of tmt11, or to specify tmt18 instead of tmt16" Reqd="No" /><Param Section="IonQuant" Name="FeatureDetectionMZTolerance" Value="10" Reqd="No" /><Param Section="IonQuant" Name="FeatureDetectionRTTolerance" Value="0.4" Reqd="No" /><Param Section="IonQuant" Name="MbrMinimumCorrelation" Value="0" Reqd="No" /><Param Section="IonQuant" Name="MbrRTTolerance" Value="1" Reqd="No" /><Param Section="IonQuant" Name="MbrIonFdr" Value="0.01" Reqd="No" /><Param Section="IonQuant" Name="MbrPeptideFdr" Value="1" Reqd="No" /><Param Section="IonQuant" Name="MbrProteinFdr" Value="1" Reqd="No" /><Param Section="IonQuant" Name="NormalizeIonIntensities" Value="1" Reqd="No" /><Param Section="IonQuant" Name="MinIonsForProteinQuant" Value="2" Reqd="No" />		1	1	0
\.


--
-- PostgreSQL database dump complete
--

